File: kubernetes/dockerfiles-fixed/Dockerfile.java
FROM openjdk:17-slim

# Install build tools and SpotBugs
RUN apt-get update && apt-get install -y \
    wget \
    unzip \
    maven \
    gradle \
    && rm -rf /var/lib/apt/lists/*

# Install SpotBugs
ENV SPOTBUGS_VERSION=4.8.3
RUN wget -q https://github.com/spotbugs/spotbugs/releases/download/${SPOTBUGS_VERSION}/spotbugs-${SPOTBUGS_VERSION}.zip \
    && unzip -q spotbugs-${SPOTBUGS_VERSION}.zip -d /opt \
    && rm spotbugs-${SPOTBUGS_VERSION}.zip \
    && ln -s /opt/spotbugs-${SPOTBUGS_VERSION}/bin/spotbugs /usr/local/bin/spotbugs \
    && chmod +x /usr/local/bin/spotbugs

# Install PMD
ENV PMD_VERSION=7.0.0
RUN wget -q https://github.com/pmd/pmd/releases/download/pmd_releases%2F${PMD_VERSION}/pmd-dist-${PMD_VERSION}-bin.zip \
    && unzip -q pmd-dist-${PMD_VERSION}-bin.zip -d /opt \
    && rm pmd-dist-${PMD_VERSION}-bin.zip \
    && ln -s /opt/pmd-bin-${PMD_VERSION}/bin/pmd /usr/local/bin/pmd \
    && chmod +x /usr/local/bin/pmd

# Install Checkstyle
ENV CHECKSTYLE_VERSION=10.12.5
RUN wget -q https://github.com/checkstyle/checkstyle/releases/download/checkstyle-${CHECKSTYLE_VERSION}/checkstyle-${CHECKSTYLE_VERSION}-all.jar \
    -O /opt/checkstyle.jar

# Create wrapper script for Checkstyle
RUN echo '#!/bin/bash\njava -jar /opt/checkstyle.jar "$@"' > /usr/local/bin/checkstyle \
    && chmod +x /usr/local/bin/checkstyle

WORKDIR /workspace

CMD ["/bin/bash"]

File: kubernetes/dockerfiles-fixed/Dockerfile.php
FROM php:8.2-cli

# Install system dependencies
RUN apt-get update && apt-get install -y \
    git \
    unzip \
    libzip-dev \
    && docker-php-ext-install zip \
    && rm -rf /var/lib/apt/lists/*

# Install Composer
RUN php -r "copy('https://getcomposer.org/installer', 'composer-setup.php');" \
    && php composer-setup.php --install-dir=/usr/local/bin --filename=composer \
    && php -r "unlink('composer-setup.php');"

# Create a temporary directory for global packages
WORKDIR /tmp/php-tools

# Install PHP analysis tools globally
RUN composer require --dev \
    phpstan/phpstan:^1.10 \
    phpmd/phpmd:^2.14 \
    squizlabs/php_codesniffer:^3.7 \
    vimeo/psalm:^5.15 \
    phan/phan:^5.4 \
    && mv vendor /opt/php-tools \
    && rm -rf /tmp/php-tools

# Create symlinks for the tools
RUN ln -s /opt/php-tools/bin/phpstan /usr/local/bin/phpstan \
    && ln -s /opt/php-tools/bin/phpmd /usr/local/bin/phpmd \
    && ln -s /opt/php-tools/bin/phpcs /usr/local/bin/phpcs \
    && ln -s /opt/php-tools/bin/phpcbf /usr/local/bin/phpcbf \
    && ln -s /opt/php-tools/bin/psalm /usr/local/bin/psalm \
    && ln -s /opt/php-tools/bin/phan /usr/local/bin/phan

# Create default PHPStan configuration
RUN echo 'parameters:\n\
  level: 5\n\
  paths:\n\
    - .\n\
  excludePaths:\n\
    - vendor\n\
    - tests\n\
  treatPhpDocTypesAsCertain: false\n\
  reportUnmatchedIgnoredErrors: false\n\
  checkMissingIterableValueType: false' > /etc/phpstan.neon

# Create default PHPMD ruleset
RUN echo '<?xml version="1.0"?>\n\
<ruleset name="Default PHP Mess Detector Ruleset"\n\
         xmlns="http://pmd.sf.net/ruleset/1.0.0"\n\
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"\n\
         xsi:schemaLocation="http://pmd.sf.net/ruleset/1.0.0 http://pmd.sf.net/ruleset_xml_schema.xsd"\n\
         xsi:noNamespaceSchemaLocation="http://pmd.sf.net/ruleset_xml_schema.xsd">\n\
  <description>Default ruleset for PHP Mess Detector</description>\n\
  <rule ref="rulesets/codesize.xml" />\n\
  <rule ref="rulesets/controversial.xml" />\n\
  <rule ref="rulesets/design.xml" />\n\
  <rule ref="rulesets/naming.xml" />\n\
  <rule ref="rulesets/unusedcode.xml" />\n\
</ruleset>' > /etc/phpmd.xml

WORKDIR /workspace

CMD ["/bin/bash"]